tutorial/02_segmentation_and_measuring.md

```diff
@@ -1,494 +1,598 @@
-  *  Thresholding algorithms which require user input (`skimage.filters.threshold_*`)
-  * `skimage.segmentation.random_walker`
-  * `skimage.segmentation.active_contour`
-  * `skimage.segmentation.watershed`
-* **Unsupervised** segmentation: No prior knowledge.  These algorithms attempt to subdivide into meaningful regions automatically.  The user may be able to tweak settings like number of regions.
-  *  Thresholding algorithms which require no user input.
-  *  `skimage.segmentation.slic`
-  * `skimage.segmentation.chan_vese`
-  * `skimage.segmentation.felzenszwalb`
-  * `skimage.segmentation.quickshift`
+Segmentation is a fundamental operation in scientific image analysis because we often want to measure properties of real, physical *objects* such as cells embedded in our image. As such, we want to find those objects within our image.
 
 +++
 
-First, some standard imports and a helper function to display our results
+Computationally, segmentations are most often represented as images, of the same size as the original image, containing integer *labels*, with one value representing one object.
+
+Here is a very simple image and segmentation, taken from [this scikit-image tutorial](https://scikit-image.org/docs/dev/auto_examples/segmentation/plot_watershed.html#sphx-glr-auto-examples-segmentation-plot-watershed-py):
 
 ```{code-cell} ipython3
 import numpy as np
-import matplotlib.pyplot as plt
+from scipy import ndimage as ndi
 
-import skimage.data as data
-import skimage.segmentation as seg
-from skimage import filters
-from skimage import draw
-from skimage import color
-from skimage import exposure
+import napari
+
+from skimage.segmentation import watershed
+from skimage.feature import peak_local_max
 
 
-def image_show(image, nrows=1, ncols=1, cmap='gray', **kwargs):
-    fig, ax = plt.subplots(nrows=nrows, ncols=ncols, figsize=(16, 16))
-    ax.imshow(image, cmap='gray')
-    ax.axis('off')
-    return fig, ax
+# Generate an initial image with two overlapping circles
+x, y = np.indices((80, 80))
+x1, y1, x2, y2 = 28, 28, 44, 52
+r1, r2 = 16, 20
+mask_circle1 = (x - x1)**2 + (y - y1)**2 < r1**2
+mask_circle2 = (x - x2)**2 + (y - y2)**2 < r2**2
+image = np.logical_or(mask_circle1, mask_circle2)
+
+# Now we want to separate the two objects in image
+# Generate the markers as local maxima of the distance to the background
+distance = ndi.distance_transform_edt(image)
+coords = peak_local_max(distance, footprint=np.ones((3, 3)), labels=image)
+mask = np.zeros(distance.shape, dtype=bool)
+mask[tuple(coords.T)] = True
+markers, _ = ndi.label(mask)
+labels = watershed(-distance, markers, mask=image)
+
+viewer = napari.Viewer()
+image_layer = viewer.add_image(image)
+labels_layer = viewer.add_labels(labels)
+labels_as_image_layer = viewer.add_image(
+    labels, name='labels as image'
+)
```

Comment:
When running this cell, Napari doesn't display the image fully. Instead part of the viewport is just black. I can move the image around as expected but it's only visible in a part of the viewport:

![image](https://github.com/scipy-2023-image-analysis/tutorial/assets/20140352/bc00ba9d-40d9-445d-8f10-80536838fc67)

Quick searching doesn't immediately give an answer. Have you seen this before? Perhaps related to HiDPI on Linux messing stuff up?

Reply:
Hmm, even stranger. The different layers show beside each other and are not stacked: 
![image](https://github.com/scipy-2023-image-analysis/tutorial/assets/20140352/92d2dae9-ed97-4f9e-a320-a022c3fa8ca7)

Reply:
<details>

```
napari: 0.4.18rc2
Platform: Linux-6.3.9-arch1-1-x86_64-with-glibc2.37
System: Arch Linux
Python: 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:40:32) [GCC 12.3.0]
Qt: 5.15.6
PyQt5: 5.15.7
NumPy: 1.25.0
SciPy: 1.11.1
Dask: 2023.6.1
VisPy: 0.12.2
magicgui: 0.7.2
superqt: unknown
in-n-out: 0.1.7
app-model: 0.1.4
npe2: 0.7.0

OpenGL:
- GL version: 4.6 (Compatibility Profile) Mesa 23.1.3
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1707x960, scale 1.5

Settings path:
- /home/lg/.config/napari/image-analysis-23_7f0765db03aaec75ab3cdd8b32a00191e1ef37d1/settings.yaml

```

</details>

Reply:
Damn, it works when I set the global scaling in Plasma / X11 back to 100%. So it's a me issue. :pray: Inconvenient for me.

Reply:
Wayland to the rescue. :magic_wand: Not perfect but would work if necessary during the tutorial. (Sorry for the noise)

Reply:
Found this: https://github.com/napari/napari/issues/1417 :smile: Stéfan seemed to have the exact same issue with cropping.